### PR TITLE
feat: chatting to contacts (multiple contacts and specific contact id) with redesign

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
This PR aims to add the chatting feature e.g. single, multiple and specific contact id usecases. Single messaging was already merged previously, this PR will fix the entire ui in sample app.

## Links
- Redesign work of previously merged v of MINI-1651
- MINI-2470
- MINI-2473

## Screenshot
| Single contact picker | Specific contact | Multiple contacts picker |
|---------------|---------------|------------------|
|Screen Shot 2021-04-08 at 1 58 54|Screen Shot 2021-04-08 at 2 05 54|Screen Shot 2021-04-08 at 2 02 35|

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/chat/ChatWindow.kt [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        contactSelectionAlertDialog.show()
+    }
+
+    fun openSpecificContactIdSelection(
+        contactId: String?,
+        message: MessageToContact,
+        onSuccess: (contactId: String?) -> Unit,
+        onError: (message: String) -> Unit
+    ) {
+        checkContactAvailability()
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/chat/ChatWindow.kt:80]: Ah, I think, I don't need to check here.

File: miniapp/USERGUIDE.md [hunk 2]
```diff
@@ -538,12 +571,6 @@ In Host App, we can get the downloaded manifest information as following:
 
 HostApp can compare between the `downloadedManifest` and the latest manifest by `MiniApp.getMiniAppManifest` to detect any new changes.
 
-### Send message to contacts
```
Comment (corycaywood) [inline miniapp/USERGUIDE.md:541]: I think we should keep this section and have the links to the API Docs for each method. 

Maybe we should also have a brief explanation of what each messaging method is used for and explain that they should show a contact picker UI?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt [hunk 3]
```diff
@@ -18,4 +18,31 @@ interface ChatBridgeDispatcher {
...
+        onSuccess: (contactId: String?) -> Unit,
+        onError: (message: String) -> Unit
+    )
+
+    /**
+     * Triggered when Mini App wants to send a message to multiple contacts.
+     * Should open a contact chooser which allows the user to choose multiple contacts,
+     * and should then send the message to all chosen contacts.
+     * Should invoke [onSuccess] with a list of IDs of the contacts which were successfully sent the message.
+     * If the user cancelled sending the message, should invoked [onSuccess] with an empty list.
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt:40]: @Climbatize @khairul-alam-licon @minh-rakuten Is this correct to return an empty list when cancelled? I thought we had discussed returning null in that case.

**Review by corycaywood — COMMENTED** [review]
It's looking good 👍  Just a couple minor comments.

**Review by minh-rakuten — APPROVED** [review]
👍 👍 
@khairul-alam-licon @Climbatize It is ready to proceed with an extra PR for layout.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt (hunk 3 above)
Comment (Climbatize) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt:40]: yep, I changed this on iOS side, now I return `null`

File: miniapp/USERGUIDE.md (hunk 2 above)
Comment (Climbatize) [reply miniapp/USERGUIDE.md:541]: You can take inspiration on the iOS PR for this @khairul-alam-licon

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt (hunk 3 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridgeDispatcher.kt]: Yeah, I have updated this part.

File: miniapp/USERGUIDE.md (hunk 2 above)
Comment (khairul-alam-licon) [reply miniapp/USERGUIDE.md:541]: True, I have updated the user guide.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridge.kt [hunk 4]
```diff
@@ -104,5 +107,6 @@ internal class ChatBridge {
 
     internal companion object {
         const val ERR_SEND_MESSAGE = "Cannot send message:"
+        const val NULL_RESPONSE = "null"
```
Comment (minh-rakuten) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/chat/ChatBridge.kt:110]: I think you can pass "null" directly or please move this into `ErrorBridgeMessage` in `MiniAppMessageBridge.kt`.

**Review by minh-rakuten — COMMENTED** [review]
There are two problems:

- The action of sending message to misleading contact id is executed successfully which is not the correct flow.

- Considering how to copy contact id easily from our sample app.

**Review by corycaywood — APPROVED** [review]
👍

**Review by minh-rakuten — APPROVED** [review]
👍

**Comment by khairul-alam-licon** [comment]
Thank you for the reviews guys,
and the UI commits from @Climbatize

**Pull request merged**